engine/flink/components/base/src/it/scala/pl/touk/nussknacker/engine/flink/util/transformer/UnionTransformerSpec.scala

```diff
@@ -35,37 +35,117 @@ class UnionTransformerSpec extends FunSuite with BeforeAndAfterAll with Matchers
+      GraphBuilder.source("start-foo", "source")
+        .branchEnd(BranchFooId, UnionNodeId),
+      GraphBuilder.source("start-bar", "noopSource")
+        .branchEnd(BranchBarId, UnionNodeId),
+      GraphBuilder
+        .branch(UnionNodeId, "union", Some(OutVariableName),
+          List(
+            BranchFooId -> List("value" -> "{a: #input}"),
+            BranchBarId -> List("value" -> "{a: '123'}"))
+        )
+        .processorEnd("end", "mockService", "all" -> s"#$OutVariableName.a")
+    ))
+
+    run(scenario, data)
+    MockService.data shouldBe data
+  }
+
+  test("should unify streams with union when both branches emit data") {
+    MockService.clear()
+    val scenario = EspProcess(MetaData("sample-union", StreamMetaData()), NonEmptyList.of[SourceNode](
+      GraphBuilder.source("start-foo", "source")
+        .branchEnd(BranchFooId, UnionNodeId),
+      GraphBuilder.source("start-bar", "source")
+        .branchEnd(BranchBarId, UnionNodeId),
+      GraphBuilder
+        .branch(UnionNodeId, "union", Some(OutVariableName),
+          List(
+            BranchFooId -> List("value" -> "{a: #input}"),
+            BranchBarId -> List("value" -> "{a: '123'}"))
+        )
+        .processorEnd("end", "mockService", "all" -> s"#$OutVariableName.a")
+    ))
+
+    run(scenario, data)
+    MockService.data.size shouldBe data.size * 2
+    MockService.data.toSet shouldBe data.toSet + "123"
+  }
+
+  test("should throw when contexts are different") {
+    MockService.clear()
+    val scenario = EspProcess(MetaData("sample-union", StreamMetaData()), NonEmptyList.of[SourceNode](
+      GraphBuilder.source("start-foo", "source")
+        .branchEnd(BranchFooId, UnionNodeId),
+      GraphBuilder.source("start-bar", "noopSource")
+        .branchEnd(BranchBarId, UnionNodeId),
+      GraphBuilder
+        .branch(UnionNodeId, "union", Some(OutVariableName),
+          List(
+            BranchFooId -> List("value" -> "{a: #input}"),
+            BranchBarId -> List("value" -> "{b: 123}")
+          )
+        )
+        .processorEnd("end", "mockService", "all" -> s"#$OutVariableName.a")
+    ))
+
+    intercept[IllegalArgumentException] {
+      run(scenario, data)
+    }.getMessage should include("All branch values must be of the same")
+  }
+
+  test("should not throw when one branch emits error") {
+    MockService.clear()
+    val scenario = EspProcess(MetaData("sample-union", StreamMetaData()), NonEmptyList.of[SourceNode](
+      GraphBuilder.source("start-foo", "source")
+        .branchEnd(BranchFooId, UnionNodeId),
+      GraphBuilder.source("start-bar", "source")
+        .branchEnd(BranchBarId, UnionNodeId),
+      GraphBuilder
+        .branch(UnionNodeId, "union", Some(OutVariableName),
+          List(
+            BranchFooId -> List("value" -> "#CONV.toNumber(#input).intValue"),
+            BranchBarId -> List("value" -> "#CONV.toNumber(#input).intValue / (#CONV.toNumber(#input).intValue % 4)")
```

Comment:
let's use sth like `#input.length - 2` (or input == 0) so we can have simpler setup, without CONF

Reply:
removed conv dependency, done it other way